internal/provider/kubernetes/httproute.go

```diff
@@ -121,17 +121,58 @@ func (r *httpRouteReconciler) Reconcile(ctx context.Context, request reconcile.R
 
 	log.Info("reconciling httproute")
 
-	// Fetch the HTTPRoute from the cache.
-	httpRoute := &gwapiv1b1.HTTPRoute{}
-	err := r.client.Get(ctx, request.NamespacedName, httpRoute)
-	if errors.IsNotFound(err) {
-		log.V(2).Info("httproute not found, deleting it from the ResourceTable")
+	// Fetch all HTTPRoutes from the cache.
+	routeList := &gwapiv1b1.HTTPRouteList{}
+	if err := r.client.List(ctx, routeList); err != nil {
+		return reconcile.Result{}, fmt.Errorf("error listing httproutes")
+	}
+
+	found := false
```

Comment:
looks like this is never getting set to true, you do still need to keep track of if you find the item that triggered the Reconcile request so that the delete logic below only runs if that item no longer exists.

Reply:
@skriss thanks for catching this. I need to slow down and double-check my work ;-) I also updated the Kube provider integration test to ensure the route's ns is being added to the watchable map.